src/rfq/protocols/bebop/client.rs

```diff
@@ -0,0 +1,510 @@
+                        tracing::error!("Failed to connect to Bebop WebSocket (attempt {}): {}", reconnect_attempts, e);
+
+                        if reconnect_attempts >= MAX_RECONNECT_ATTEMPTS {
+                            yield Err(RFQError::ConnectionError(format!("Failed to connect after {MAX_RECONNECT_ATTEMPTS} attempts: {e}")));
+                            return;
+                        }
+
+                        let backoff_duration = Duration::from_secs(2_u64.pow(reconnect_attempts.min(5)));
+                        tracing::info!("Retrying connection in {} seconds...", backoff_duration.as_secs());
+                        sleep(backoff_duration).await;
+                        continue;
+                    }
+                };
+
+                let (_, mut ws_receiver) = ws_stream.split();
+
+                // Message processing loop
+                while let Some(msg) = ws_receiver.next().await {
+                    match msg {
+                        Ok(Message::Text(text)) => {
+                            match serde_json::from_str::<BebopPriceMessage>(&text) {
+                                Ok(price_data_map) => {
+                                    let mut new_components = HashMap::new();
+                                    let mut latest_timestamp = 0u64;
+
+                                    // Process all pairs from this WebSocket message
+                                    for (pair, price_data) in price_data_map {
+                                        if pairs.contains(&pair) {
+                                            let tvl = client.calculate_tvl(&price_data);
+                                            if tvl < tvl_threshold {
+                                                continue;
+                                            }
+
+                                            let component_id = format!("bebop_{}", pair.replace("/", "_"));
+
+                                            let tokens;
+                                            if let Some((token0, token1)) = pair.split_once('/') {
+                                                tokens = vec![token0.as_bytes().to_vec().into(), token1.as_bytes().to_vec().into()]
+                                            } else {
+                                                continue;
+                                            };
+
+                                            let component_with_state = client.create_component_with_state(component_id.clone(), tokens, &price_data, tvl);
+                                            new_components.insert(component_id, component_with_state);
+
+                                            // Track the latest timestamp across all pairs
+                                            let timestamp = price_data.last_update_ts as u64;
+                                            if timestamp > latest_timestamp {
+                                                latest_timestamp = timestamp;
```

Comment:
why?

Reply:
Because the timestamp differers per pair... and we can only have one timestamp in the `StateSyncMessage`? Or where do you suggest getting this timestamp? Just taking the current timestamp?

Reply:
yeah actually I think the current timestamp is ok. it's the timestamp we receive the data